### DISCUSSION TITLE
Extend the width of a column within a table

### DISCUSSION THREAD

**Discussion by joebudi** [Budibase/budibase#1694] [Ideas]
As a developer, I would like to extend the width of a column within a table, so that I can present longer content to users.

**Comment by mjashanks** [comment]
Bumping this up as it's been requested a few times: 

- https://github.com/Budibase/budibase/discussions/2179
- https://github.com/Budibase/budibase/discussions/1861
- https://github.com/Budibase/budibase/discussions/2749

**Reply by aptkingston** [reply]
Been chatting to Joe about this but writing things up here for reference. I see 2 improvements which can be made here, which can be implemented together or independently.

1. If the table does not scroll then we can automatically size everything, which means that long fields will take up more room and short fields will take up less room. This makes everything much more readable. There will need to be a sensible max width applied to every column to avoid stupidly long columns. Like I said, this will only work if the table does not scroll, so it's best suited for pagination which is the default now anyway.
2. Allow manual sizing of columns. I see this being combined with some other features to give a better table experience overall. A new table configuration component will be made that opens in a drawer, and looks similar to how navigation links are configured. This component will allow table columns to be added, removed and reordered, and will also allow setting a width and a label for the column, if desired. That solves a couple of issues in one go, and it makes sense for them all to be configured together.
If you specify a fixed width for some columns but not for others, then the other columns will be treated as they are today, unless solution 1 is also implemented which would automatically size any columns without a fixed width when the table does not scroll.
